addons/event_sale/models/event_registration.py

```diff
@@ -46,6 +54,12 @@ def write(self, vals):
                 self.env['sale.order.line'].browse(vals['sale_order_line_id'])
             )
             vals.update(so_line_vals)
+
+        if vals.get('event_ticket_id'):
+            new_ticket_id = self.env['event.event.ticket'].browse(vals['event_ticket_id'])
```

Comment:
Could test the ids before browsing the model to avoid unnecessary requests.

```
if self.event_ticket_id and self.event_ticket_id.id != vals['event_ticket_id']:
    self._sale_order_ticket_type_change_notify(self.env['event.event.ticket'].browse(vals['event_ticket_id']))
```